test/support/otlp_endpoint_test.rb

```diff
@@ -0,0 +1,287 @@
+    end
+
+    # 5
+    it 'OTEL METRICS ENDPOINT to special and no SW_APM_COLLECTOR' do
+      ENV['OTEL_EXPORTER_OTLP_METRICS_ENDPOINT'] = 'http://special.host:4317/v1/metrics'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint_types.each { |data_type| endpoint.configure_otlp_endpoint(data_type) }
+
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_METRICS_ENDPOINT', nil)).must_equal 'http://special.host:4317/v1/metrics'
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443/v1/traces'
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_LOGS_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443/v1/logs'
+      _(ENV.fetch('SW_APM_COLLECTOR', nil)).must_equal 'apm.collector.na-01.cloud.solarwinds.com'
+    end
+
+    it 'OTEL METRICS ENDPOINT to special and SW_APM_COLLECTOR to special location' do
+      ENV['OTEL_EXPORTER_OTLP_METRICS_ENDPOINT'] = 'http://special.host:4317/v1/metrics'
+      ENV['SW_APM_COLLECTOR'] = 'apm.collector.eu-01.cloud.solarwinds.com'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint_types.each { |data_type| endpoint.configure_otlp_endpoint(data_type) }
+
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_METRICS_ENDPOINT', nil)).must_equal 'http://special.host:4317/v1/metrics'
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', nil)).must_equal 'https://otel.collector.eu-01.cloud.solarwinds.com:443/v1/traces'
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_LOGS_ENDPOINT', nil)).must_equal 'https://otel.collector.eu-01.cloud.solarwinds.com:443/v1/logs'
+      _(ENV.fetch('SW_APM_COLLECTOR', nil)).must_equal 'apm.collector.eu-01.cloud.solarwinds.com'
+    end
+
+    # 7,8 not appliable since SW_APM_LEGACY will not be present here
+  end
+
+  describe 'config_service_name' do
+    it 'resource attribute set swo apm service key set' do
+      ENV['OTEL_RESOURCE_ATTRIBUTES'] = 'sw.apm.version=1.1.1,sw.data.module=apm,service.name=otel-autodetected-default'
+      ENV['SW_APM_SERVICE_KEY'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234:my-service'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_service_name
+
+      _(endpoint.instance_variable_get(:@service_name)).must_equal 'otel-autodetected-default'
+    end
+
+    it 'otel service name set resource attribute set swo apm service key set' do
+      ENV['OTEL_RESOURCE_ATTRIBUTES'] = 'sw.apm.version=1.1.1,sw.data.module=apm,service.name=otel-autodetected-default'
+      ENV['OTEL_SERVICE_NAME'] = 'otel-service-name'
+      ENV['SW_APM_SERVICE_KEY'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234:my-service'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_service_name
+
+      _(endpoint.instance_variable_get(:@service_name)).must_equal 'otel-service-name'
+    end
+
+    it 'otel service name set resource attribute set' do
```

Comment:
i don't see this test case as effectively different from the above one.

Reply:
Maybe remove this test case; since it's mainly test that if service key is invalid, then agent should be disabled.